server/auth.js

```diff
@@ -19,12 +19,31 @@ if (!process.env.GOOGLE_APPLICATION_CREDENTIALS && process.env.NODE_ENV === 'dev
   process.env.GOOGLE_APPLICATION_CREDENTIALS = path.join(__dirname, '.auth.json')
 }
 
-// only public method, returns the authClient that can be used for making other requests
+// returns the authClient that can be used for making other requests
 exports.getAuth = async () => {
   if (authClient && process.env.NODE_ENV !== 'test') return authClient
   return setAuthClient()
 }
 
+// This is only used when data is fetched manually via a download link vs using the Google Drive library
+exports.getAccessToken = async () => {
+  const serviceAccountKey = process.env.GOOGLE_APPLICATION_JSON
+  const auth = new google.auth.GoogleAuth({
+    credentials: serviceAccountKey,
+    scopes: 'https://www.googleapis.com/auth/drive.readonly'
+  })
+
+  const accessToken = await auth.getClient()
```

Comment:
So this `auth.getClient()` is a different client than the `authClient` that is set up in `setAuthClient` below. Is it because of different `scopes`? Would there be any reasonable way to use same client for all use cases? And to avoid creating a new client every time this function is called?

Reply:
Updated [here](https://github.com/reaktor/library/pull/7/commits/ee9c3bf44e17f15c292f544636512eea422a556a)